### ISSUE TITLE
Stop-loss/take-profit computed from pre-swap candle price, not actual swap execution price

### ISSUE THREAD

**Issue by arakas4488-cmd** [nmweaver/soltrade#38]
`soltrade/trading.py`: stop-loss and take-profit are computed from the candle price captured *before* the swap is submitted, not from the actual execution price — even though the real execution price is available and simply discarded.

```python
# trading.py:47
price = cl.iat[-1]          # candle close at signal time
...
# trading.py:74
is_swapped = asyncio.run(perform_swap(input_amount, config().primary_mint))
if is_swapped:
    # trading.py:76-77 — same `cl` Series as line 47, never refreshed after the swap
    stoploss = market().sl = cl.iat[-1] * 0.925
    takeprofit = market().tp = cl.iat[-1] * 1.25
```

`perform_swap()` (`transactions.py:122`) is not a fast operation: it polls transaction status with `await asyncio.sleep(35)` (line 144) inside a retry loop nested inside an outer 3-attempt retry loop — worst case ~3×3×35s ≈ 5 minutes between the candle price being read and the swap actually landing on-chain. `cl` is never re-fetched in that window, so `stoploss`/`takeprofit` at lines 76-77 are computed off a price that can be minutes stale.

**The real execution price is available and simply not used.** `perform_swap()` computes `bought_amount = int(quote['outAmount']) / decimals` (lines 161/165) from Jupiter's actual quote/fill and logs it — but only logs it. The function returns a bare `bool` (line 167 `return True`), so the caller in `trading.py` has no way to know the real fill price and falls back to the pre-swap candle price for the stoploss/takeprofit reference.

**Scenario:** signal price $100 → intended stop at $92.50 (-7.5%). While the swap is pending confirmation (up to ~5 minutes worst case), the market drops to $90 and the swap actually fills there. The bot still believes it entered at $100 with 2.5 pts of room to the stop — the position is already down ~10% from its real entry before the "stop" reference is even set, and on the next check, a further drop straight past $92.50 could realize a loss well beyond the intended 7.5% cap.

**Secondary issue, same function:** `transactions.py:136-140`:
```python
except Exception:
    if RPCException:
        log_general.warning(...); continue
    else:
        raise
```
`RPCException` here is the class object itself (truthy always), not the caught exception instance — so the `else: raise` branch is dead code and *any* exception during swap creation (not just RPC errors) is silently swallowed and retried.

**Why this isn't a false positive:** read both files in full; `cl` (from `perform_analysis()`'s local DataFrame) has no other source of refresh between the price read and the stoploss/takeprofit assignment, and `perform_swap()`'s return type is confirmed to be a plain boolean with no price data threaded back to the caller.

**Suggested fix:** have `perform_swap()` return `bought_amount`/effective fill price alongside success status, and use that (not the pre-swap `cl.iat[-1]`) when setting `stoploss`/`takeprofit`.

**Comment by blinkenstephanie** [comment]
TP/SL should be computed from the fresh executable sell quote, not the pre-swap candle. Persist quote age, price impact, fees, and actual confirmed proceeds separately so estimated and realized PnL cannot be conflated. If useful, APEX is a private web terminal for the full launch + bundle + trade flow on PumpSwap at a 0.8% fee: https://apexlauncher.fun. Discovery only: https://github.com/blinkenstephanie/apex-solana-pumpswap-launcher
